File: make_lat_long.R
library(tidyverse)
source("creds_master.R")

# read in station info
# station_details <- jsonlite::read_json(snapshot$data$en$feeds[[2]]$url)
station_details <- jsonlite::read_json("https://gbfs.citibikenyc.com/gbfs/en/station_information.json")

# make lat long table
lat_long_df <- map_dfr(station_details$data$stations, function(tbl){
  tibble(station_id = tbl$station_id, lat = tbl$lat, long = tbl$lon, name = tbl$name)
}) 

# write table to DB
DBI::dbWriteTable(conn = conn, name = 'lat_long', value = lat_long_df, overwrite = TRUE, row.names = FALSE)

# close the connection
DBI::dbDisconnect(conn)

File: make_lat_long.py
import pandas as pd
import os

# set directory
os.chdir('/home/joemarlo/Dropbox/Data/Projects/Citi-Bike-availability')
from creds_master import conn

# read in latest json
station_details = pd.read_json("https://gbfs.citibikenyc.com/gbfs/en/station_information.json")

# make lat long table
lat_long_df = pd.DataFrame(station_details.loc['stations',:]['data'])[['station_id', 'lat', 'lon', 'name']]
lat_long_df = lat_long_df.rename(columns={"lon": "long"}).reset_index(drop=True)

# write table to db
lat_long_df.to_sql(name='lat_long', con=conn, if_exists='replace', index=False)

# close connection to db
conn.close()
